src/codegen.cpp

```diff
@@ -299,8 +298,15 @@ int32_t jl_jlcall_api(const void *function)
     // give the function an index in the constant lookup table
     if (function == NULL)
         return 0;
-    const Function *F = (const Function*)function;
-    return (F->getFunctionType() == jl_func_sig ? 1 : 3);
+    const Function *F = cast<const Function>((const Value*)function);
+    StringRef Name = F->getName();
+    if (Name.startswith("japi3_"))
+        return 3;
+    assert(Name.startswith("japi1_") ||
```

Comment:
Julia ABI N would perhaps be the most accurate. But these are not externally visible names.